rollup/fees/rollup_fee.go

```diff
@@ -210,6 +210,33 @@ func estimateTxCompressionRatio(data []byte, blockNumber uint64, blockTime uint6
 	return ratio, nil
 }
 
+// calculateTxCompressedSize calculates the size of `data` after compression using da-codec.
+// We constrain compressed_size so that it cannot exceed the original size:
+//
+//	compressed_size(tx) = min(size(zstd(rlp(tx))), size(rlp(tx)))
+//
+// This provides an upper bound on the rollup fee for a given transaction, regardless
+// what compression algorithm the sequencer/prover uses.
+func calculateTxCompressedSize(data []byte, blockNumber uint64, blockTime uint64, config *params.ChainConfig) (*big.Int, error) {
+	// Compressed size of empty data is 0.
+	// In practice, the rlp-encoded transaction is always non-empty.
+	if len(data) == 0 {
+		return common.Big0, nil
+	}
+
+	// Compress data using da-codec
+	compressed, err := encoding.CompressScrollBatchBytes(data, blockNumber, blockTime, config)
```

Comment:
Just as a reminder: we need to watch out in case we change the zstd compression level here for batch submission